packages/client/src/links/wsLink.ts

```diff
@@ -212,11 +212,15 @@ export function createWSClient(opts: WebSocketClientOptions) {
         if (req.ws !== conn) {
           continue;
         }
-        req.callbacks.error?.(
-          TRPCClientError.from(
-            new TRPCWebSocketClosedError('WebSocket closed prematurely'),
-          ),
-        );
+
+        if (state === 'closed') {
```

Comment:
This should be the _desired_ state, so we should log here if it's _not_ closed

```suggestion
        if (state !== 'closed') {
```

Reply:
Actually I think we should always wait here.... we shouldn't try to close the connection if there are pending requests